code/addons/controls/src/ControlsPanel.tsx

```diff
@@ -21,24 +16,26 @@ interface ControlsParameters {
   sort?: SortType;
   expanded?: boolean;
   presetColors?: PresetColor[];
+
+  /** @deprecated No longer used, will be removed in Storybook 8.0 */
   hideNoControlsWarning?: boolean;
 }
 
 export const ControlsPanel: FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
   const [args, updateArgs, resetArgs] = useArgs();
   const [globals] = useGlobals();
   const rows = useArgTypes();
-  const isArgsStory = useParameter<boolean>('__isArgsStory', false);
-  const {
-    expanded,
-    sort,
-    presetColors,
-    hideNoControlsWarning = false,
-  } = useParameter<ControlsParameters>(PARAM_KEY, {});
-  const { path } = useStorybookState();
+  const { expanded, sort, presetColors } = useParameter<ControlsParameters>(PARAM_KEY, {});
+  const { path, previewInitialized } = useStorybookState();
+
+  // If the story is prepared, then show the args table
+  // and reset the loading states
+  useEffect(() => {
+    if (previewInitialized) setIsLoading(false);
+  }, [previewInitialized]);
```

Comment:
I'm honestly not sure what happens if you call `setState` with the same state that it already has.

```suggestion
  useEffect(() => {
    if (previewInitialized && isLoading) setIsLoading(false);
  }, [previewInitialized]);
```

Reply:
make sure to add isLoading and setIsLoading to the hook dependencies array too

Reply:
We agreed not to do that @yannbf. It's working without it.